### PR TITLE
Feature/#227 import coco

### LINKED ISSUE #227: Allow importing COCO labels/dataset to LabelFlow
**Describe the solution you'd like (import visuals when possible)**
- [x] Import modal: accept zip file corresponding to the COCO export zip file
  - [x] Add images to the opened dataset
  - [x] Create classes
  - [x] Add labels to the imported images
- [x] Zip file are imported assuming they it's a COCO export file with images + COCO Json

Blocked by #226

### PR DESCRIPTION
**Pull request by etiennedupont**
# Feature

## Work performed

- Created a new resolver named `importDataset` (counterpart of `exportDataset`)
- Enabled zip and json files in the dropzone of the "import images" modal
- Whether you uploaded a json or a zip file is an option the client give to the resolver
- The resolver skips images and labelclasses that are already in the dataset, check is made on file name
- The resolver infers whether the annotations are boxes or polygons (no other way since coco doesn't provide type information)

## Caveats

- I had to copy `typescript/web/src/utils/class-color-generator.ts` to the `common-resolvers` utilities to enable proper colouring of the classes in the resolver
- No granular tracking of the status of the import, it's binary for the whole import

## Resolved issues

Closes #227

### CODE REVIEW

**Comment by etiennedupont** [comment]
Import from zip use case:

sr1

**Comment by etiennedupont** [comment]
Import from annotation file use case:

sr2

File: typescript/common-resolvers/src/import/format-coco/index.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    const imagesCoco = annotationFile.images.filter(
+      (imageCoco) =>
+        !images.find(
+          (image) =>
+            image.name ===
+            imageCoco.file_name.replace(path.extname(imageCoco.file_name), "")
+        )
+    );
+    // eslint-disable-next-line no-restricted-syntax
+    for (const imageCoco of imagesCoco) {
```
Comment (nacho-carnicero) [inline typescript/common-resolvers/src/import/format-coco/index.ts:88]: Why using a `for` here?

File: typescript/common-resolvers/src/import/format-coco/index.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      cocoCategoryIdToLabelFlowLabelClassId.set(
+        categoryCoco.id,
+        labelFlowLabelClass.id
+      );
+      return false;
+    }
+    return true;
+  });
+  // eslint-disable-next-line no-restricted-syntax
+  for (const categoryCoco of categoriesCoco) {
```
Comment (nacho-carnicero) [inline typescript/common-resolvers/src/import/format-coco/index.ts:131]: Same comment for the loop

File: typescript/common-resolvers/src/import/format-coco/index.ts (hunk 1 above)
Comment (etiennedupont) [reply typescript/common-resolvers/src/import/format-coco/index.ts:88]: I need to await image creation sequentially, and I didn't want to import bluebird for that...  Maybe you'll have an alternative option?

Comment (nacho-carnicero) [reply typescript/common-resolvers/src/import/format-coco/index.ts:88]: Why does it need to be sequential?

Comment (etiennedupont) [reply typescript/common-resolvers/src/import/format-coco/index.ts:88]: To ensure that `export ∘ import = Identity`

**Review by nacho-carnicero — APPROVED** [review]

**Pull request merged**